### PR TITLE
lightning: optimize a query to I_S.columns

### LINKED ISSUE #57115: split lightning's I_S.columns query into small ones to avoid slow query
## Enhancement

currently

https://github.com/pingcap/tidb/blob/aa83e4e0d5fbba3aa502abd99bbcddc5f85f1e5a/pkg/lightning/backend/tidb/tidb.go#L193-L198

will query for the whole database. We can only query needed tables and use proper batch size

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #57115

Problem Summary:

### What changed and how does it work?

add `table_name IN (...)` to that query, split the table name list into batches with maximum size of 32, and execute them with maximum concurrency 8.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by CbcWestwolf — APPROVED** [review]

File: pkg/lightning/backend/tidb/tidb.go [hunk 1]
```diff
@@ -171,123 +171,154 @@ func (b *targetInfoGetter) FetchRemoteDBModels(ctx context.Context) ([]*model.DB
...
+			return s.Transact(
+				egCtx, "fetch table columns",
+				func(_ context.Context, tx *sql.Tx) error {
+					args := make([]any, 0, 1+end-start)
+					args = append(args, schemaName)
+					for _, tableName := range tableNames[start:end] {
+						args = append(args, tableName)
+					}
+					//nolint:gosec
+					rows, e := tx.Query(`
```
Comment (D3Hunter) [inline pkg/lightning/backend/tidb/tidb.go:211]:
```suggestion
					rows, err := tx.Query(`
```
prefer following the convention to use the name `err`. below too

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**